### PR TITLE
Add class for accessing Ansible configuration

### PR DESCRIPTION
**Pull request by ssbarnea**
This enables easy access to Ansible configuration.

### CODE REVIEW

**Comment by ssbarnea** [comment]
@webknjaz Any change to give me hand with error below:
```
src/ansible_compat/config.py:81: error: Missing type parameters for generic type "UserDict"
```

I am not really sure how to subclass UserDict and keep mypy happy.... I did google a lot but still no clear solution yet.

File: src/ansible_compat/config.py [hunk 1]
```diff
@@ -77,5 +84,50 @@ def ansible_version(version: str = "") -> Version:
...
+        config = subprocess.check_output(
+            ["ansible-config", "dump"], universal_newlines=True, env=env
+        )
+        for match in re.finditer(
+            r"^(?P<key>[A-Za-z0-9_]+).* = (?P<value>.*)$", config, re.MULTILINE
+        ):
+            key = match.groupdict()['key']
+            value = match.groupdict()['value']
+            try:
+                self[key] = eval(value)  # pylint: disable=eval-used
```
Comment (felixfontein) [inline src/ansible_compat/config.py:114]: I would really avoid using `eval` in anything than testing code.

File: src/ansible_compat/config.py [hunk 2]
```diff
@@ -77,5 +84,50 @@ def ansible_version(version: str = "") -> Version:
...
+
+    def __init__(self) -> None:
+        """Load config dictionary."""
+        super().__init__()
+        env = os.environ.copy()
+        # Avoid possible ANSI garbage
+        env["ANSIBLE_FORCE_COLOR"] = "0"
+
+        config = subprocess.check_output(
+            ["ansible-config", "dump"], universal_newlines=True, env=env
```
Comment (felixfontein) [inline src/ansible_compat/config.py:106]: How about doing this differently:
1. Have a class `AnsibleConfig` which takes the stdout of `ansible-config dump`.
2. Have a separate function that calls `ansible-config dump` and creates `AnsibleConfig` from the result.

This makes it possible to use this class in situations where you want to execute `ansible-config` in a specific venv, or want to execute it asynchronously (and not with `subprocess`).

File: src/ansible_compat/config.py (hunk 1 above)
Comment (ssbarnea) [reply src/ansible_compat/config.py:114]: Believe me that I did not want to use eval either but I did not know exactly what to use. I am open to suggestions here.

Comment (ssbarnea) [reply src/ansible_compat/config.py:114]: I added #11 for tracking this bit in follow-up.

File: src/ansible_compat/config.py (hunk 2 above)
Comment (ssbarnea) [reply src/ansible_compat/config.py]: In fact I was planning to do this in follow-up, but now that I mentioned it I just did it. Note that I did not split yet the call to ansible, we will do this later with some king of ansible-runner class that will allow us to run various ansible commands, not only galaxy install.

**Comment by ssbarnea** [comment]
Please review, I think i sorted all comments.

File: src/ansible_compat/config.py (hunk 2 above)
Comment (webknjaz) [reply src/ansible_compat/config.py]: I'd also add that this "function" might as well be a `@classmethod` — it's a common tool for custom initializers.

**Pull request merged**